crates/primitives/src/transaction/access_list.rs

```diff
@@ -81,32 +89,69 @@ impl AccessList {
         self.0.iter().map(AccessListItem::size).sum::<usize>() +
             self.0.capacity() * mem::size_of::<AccessListItem>()
     }
+
+    /// Returns the position of the given address in the access list, if present.
+    pub fn index_of_address(&self, address: Address) -> Option<usize> {
+        self.0.iter().position(|item| item.address == address)
+    }
+
+    /// Checks if a specific storage slot within an account is present in the access list.
+    ///
+    /// Returns a tuple with flags for the presence of the account and the slot.
+    pub fn contains(&self, address: Address, slot: B256) -> (bool, bool) {
+        self.index_of_address(address)
+            .map_or((false, false), |idx| (true, self.contains_storage_key_at_index(slot, idx)))
+    }
+
+    /// Checks if the access list contains the specified address.
+    pub fn contains_address(&self, address: Address) -> bool {
+        self.0.iter().any(|item| item.address == address)
+    }
+
+    /// Checks if the storage keys at the given index within an account are present in the access
+    /// list.
+    pub fn contains_storage_key_at_index(&self, slot: B256, index: usize) -> bool {
+        self.0[index].storage_keys.iter().any(|storage_key| *storage_key == slot)
```

Comment:
this will panic if the index does not exist,
I think we want to use .get().map here

Reply:
@mattsse Yes my bad, just fixed it and I've also added `Deref` and `DerefMut` implementations for `AccessList` in order to avoid calling `self.0` each time.